tests/ignite/handlers/test_state_param_scheduler.py

```diff
@@ -414,3 +391,69 @@ def __call__(self, event_index):
     param_scheduler.attach(engine, Events.EPOCH_COMPLETED)
 
     engine.run([0] * 8, max_epochs=10)
+
+
+def test_param_scheduler_attach_exception():
+    trainer = Engine(lambda e, b: None)
+    param_name = "state_param"
+
+    setattr(trainer.state, param_name, None)
+
+    save_history = True
+    create_new = True
+
+    param_scheduler = PiecewiseLinearStateScheduler(
+        param_name=param_name,
+        milestones_values=[(0, 0.0), (10, 0.999)],
+        save_history=save_history,
+        create_new=create_new,
+    )
+
+    with pytest.raises(
+        ValueError,
+        match=r"Attribute '" + re.escape(param_name) + "' already exists in the engine.state. "
+        r"This may be a conflict between multiple handlers. "
+        r"Please choose another name.",
+    ):
+        param_scheduler.attach(trainer, Events.ITERATION_COMPLETED)
+
+
+def test_param_scheduler_attach_warning():
+    trainer = Engine(lambda e, b: None)
+    param_name = "state_param"
+    save_history = True
+    create_new = False
+
+    param_scheduler = PiecewiseLinearStateScheduler(
+        param_name=param_name,
+        milestones_values=[(0, 0.0), (10, 0.999)],
+        save_history=save_history,
+        create_new=create_new,
+    )
+
+    with pytest.warns(
+        UserWarning,
+        match=r"Attribute '" + re.escape(param_name) + "' is not defined in the engine.state. "
+        r"PiecewiseLinearStateScheduler will create it. Remove this warning by setting create_new=True.",
+    ):
+        param_scheduler.attach(trainer, Events.ITERATION_COMPLETED)
+
+
+def test_param_scheduler_with_ema_handler():
+
+    from ignite.handlers import EMAHandler
+
+    model = nn.Linear(2, 1)
+    trainer = Engine(lambda e, b: model(b))
+    data = torch.rand(100, 2)
+
+    param_name = "ema_decay"
+
+    ema_handler = EMAHandler(model)
+    ema_handler.attach(trainer, name=param_name, event=Events.ITERATION_COMPLETED)
+
+    ema_decay_scheduler = PiecewiseLinearStateScheduler(
+        param_name=param_name, milestones_values=[(0, 0.0), (10, 0.999),], save_history=True
+    )
+    ema_decay_scheduler.attach(trainer, Events.ITERATION_COMPLETED)
+    trainer.run(data, max_epochs=20)
```

Comment:
@fco-dv looks good, but can we check somehow that these values were used in EMAHandler ? 
Maybe you can mock it and instead of updating a model, just ensure that values are the expected ones

Reply:
Thanks for the hints @vfdev-5 , will try this approach!

Reply:
@vfdev-5  don't you think we need now to remove the inner parameter scheduler associated to `_get_momentum` defined in the `EMAHandler`, otherwise I guess the values used by  `_update_ema_model` would not be those of `StateParamScheduler` ?

Reply:
Yes, we can deprecate them I think. Cc @sandylaker

Reply:
@vfdev-5 @fco-dv Thanks for the cc. Yes the `_get_momentum` function should be deprecated. Once an `EMAHandler` is attached, a state parameter `momentum` will be created, and it will be scheduled by the `StateParamScheduler` if the user has specialized one. Therefore, we only need to read the `momentum` value from the engine's state when updating the ema model. I will create a PR after this one is merged to the master branch.

Reply:
@sandylaker thanks for the clarification! Seems that our "working test" involving `EMAHandler` + `StateParamScheduler` would be more relevant after your PR... What do you think @vfdev-5, maybe we could add this test after the next PR ?

Reply:
Sounds good to me !